src/network/networkd-dhcp6.c

```diff
@@ -35,6 +36,88 @@ static bool dhcp6_get_prefix_delegation(Link *link) {
                       RADV_PREFIX_DELEGATION_BOTH);
 }
 
+static bool dhcp6_has_preferred_subnet_id(Link *link) {
+        assert(link->network);
```

Comment:
i think we should return false if there's no network set, like for the dhcp6_get_prefix_delegation() call above